node/src/proxy_client/stream_handler_pool.rs

```diff
@@ -29,20 +29,28 @@ use tokio::prelude::future::{err, ok};
 use trust_dns_resolver::error::ResolveError;
 use trust_dns_resolver::lookup_ip::LookupIp;
 
+// TODO: This should be renamed to ProxyClientStreamHandlerPoolReal (or something more concise)
+// to differentiate it from the other StreamHandlerPool, which, unlike this, is an actor.
 pub trait StreamHandlerPool {
     fn process_package(&self, payload: ClientRequestPayload_0v1, paying_wallet_opt: Option<Wallet>);
 }
 
 pub struct StreamHandlerPoolReal {
     inner: Arc<Mutex<StreamHandlerPoolRealInner>>,
-    stream_adder_rx: Receiver<(StreamKey, Box<dyn SenderWrapper<SequencedPacket>>)>,
+    stream_adder_rx: Receiver<(StreamKey, StreamSenders)>,
     stream_killer_rx: Receiver<(StreamKey, u64)>,
 }
 
+#[derive(Debug)]
+pub struct StreamSenders {
+    pub writer_data: Box<dyn SenderWrapper<SequencedPacket>>,
+    pub reader_shutdown_tx: Sender<()>,
+}
+
```

Comment:
Maybe move it to a few lines above, before the `StreamHandlerPool` or wherever the `SenderWrapper` is stored.